plugins/xclient.js

```diff
@@ -50,28 +46,26 @@ exports.hook_unrecognized_command = function (next, connection, params) {
 
     // If we get here - the client is allowed to use XCLIENT
     // Process arguments
-    const args = (new String(params[1])).toLowerCase().split(/ /);
+    const args = (String(params[1])).toLowerCase().split(/ /);
     const xclient = {};
-    for (let a=0; a < args.length; a++) {
-        const match = /^([^=]+)=([^ ]+)/.exec(args[a]);
+    for (const arg of args) {
+        const match = /^([^=]+)=([^ ]+)/.exec(arg);
         if (match) {
             connection.logdebug(this, `found key=${match[1]} value=${match[2]}`);
             switch (match[1]) {
                 case 'destaddr':
                 case 'addr': {
                     // IPv6 is prefixed in the XCLIENT protocol
                     let ipv6;
+                    // FIXME: should this be assigned vs compared?
```

Comment:
useless FIXME comments. Read them. Understand the code, and then get rid of them.